### PR TITLE
Refactor the TaskManager to be separate from the QueryExecutor

### PR DESCRIPTION
**Pull request by jsternberg**

### CODE REVIEW

File: influxql/task_manager.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	// Logger to use for all logging.
+	// Defaults to discarding all log output.
+	Logger *log.Logger
+
+	// Used for managing and tracking running queries.
+	queries  map[uint64]*QueryTask
+	nextID   uint64
+	mu       sync.RWMutex
+	shutdown bool
+}
```
Comment (benbjohnson) [inline influxql/task_manager.go]: Can you add a comment to describe what the `TaskManager` is?

File: coordinator/statement_executor.go [hunk 2]
```diff
@@ -41,10 +44,10 @@ type StatementExecutor struct {
 	MaxSelectBucketsN int
 }
 
-func (e *StatementExecutor) ExecuteStatement(stmt influxql.Statement, ctx *influxql.ExecutionContext) error {
+func (e *StatementExecutor) ExecuteStatement(stmt influxql.Statement, ctx influxql.ExecutionContext) error {
```
Comment (benbjohnson) [inline coordinator/statement_executor.go]: Why did the `ctx` change from a pointer?

**Comment by benbjohnson** [comment]
What's the purpose of the change? The code change lgtm.

File: coordinator/statement_executor.go (hunk 2 above)
Comment (jsternberg) [inline coordinator/statement_executor.go]: This was as a result of a conversation with @dgnorton where we were considering the ability of the `StatementExecutor` to modify the context in some cases. We thought passing it by value would make it easier for a `StatementExecutor` to do that, although it's also possible by just doing this:

```
newCtx := *ctx
```

**Comment by jsternberg** [comment]
The purpose is to allow someone to overwrite the `ShowQueriesStatement` and `KillQueryStatement`. Previously, they were directly integrated with the `QueryExecutor` and you couldn't overwrite those commands. We need to overwrite them for distributed queries. I thought about just turning `QueryExecutor` into a `StatementExecutor` and keeping a circular reference, but I don't like the concept of circular references (even in a language with GC) so I thought it would be better to split it out so the task management could be managed by different subsystems (such as being used with RPC in clustering).

File: influxql/task_manager.go (hunk 1 above)
Comment (jsternberg) [inline influxql/task_manager.go]: Added comments to all of the methods on `TaskManager` and the struct.

**Comment by benbjohnson** [comment]
:+1:

**Pull request merged**